rust/xaynet-server/src/state_machine/phases/unmask.rs

```diff
@@ -42,53 +44,57 @@ pub struct Unmask {
 }
 
 #[async_trait]
-impl<S> Phase<S> for PhaseState<Unmask, S>
+impl<T> Phase<T> for PhaseState<Unmask, T>
 where
-    S: Storage,
+    T: Storage,
 {
     const NAME: PhaseName = PhaseName::Unmask;
 
-    async fn run(&mut self) -> Result<(), PhaseStateError> {
+    async fn process(&mut self) -> Result<(), PhaseStateError> {
         self.emit_number_of_unique_masks_metrics();
-
         let best_masks = self
             .shared
             .store
             .best_masks()
             .await
             .map_err(UnmaskStateError::FetchBestMasks)?
             .ok_or(UnmaskStateError::NoMask)?;
-
-        let global_model = self.end_round(best_masks).await?;
+        self.end_round(best_masks).await?;
 
         #[cfg(feature = "model-persistence")]
-        self.save_global_model(&global_model).await?;
-
-        self.shared
-            .store
-            .publish_proof(&global_model)
-            .await
-            .map_err(UnmaskStateError::PublishProof)?;
-
-        info!("broadcasting the new global model");
-        self.shared
-            .events
-            .broadcast_model(ModelUpdate::New(Arc::new(global_model)));
+        self.save_global_model().await?;
 
         Ok(())
     }
 
-    fn next(self) -> Option<StateMachine<S>> {
+    async fn broadcast(&mut self) -> Result<(), PhaseStateError> {
+        if let Some(ref global_model) = self.shared.state.global_model {
+            info!("broadcasting proof of the new global model");
+            self.shared
```

Comment:
I would prefer to move this into the process method so that the `broadcast` can not fail and and does not have to be `async`
but it is also fine like this

Reply:
i'll move it, it is more of a store operation than a broadcast operation after all.

this also allows us to make all other `broadcast()`s sync and infallible.